hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -98,15 +97,16 @@
   private final long lockCheckMinWaitTime;
   private final long lockCheckMaxWaitTime;
   private final FileIO fileIO;
+  private final HiveCatalog catalog;
 
-  protected HiveTableOperations(Configuration conf, HiveClientPool metaClients, FileIO fileIO,
-                                String catalogName, String database, String table) {
+  protected HiveTableOperations(Configuration conf, FileIO fileIO, HiveCatalog catalog,
```

Comment:
Along w/ my comment above I don't think the table ops should get to see the entire catalog, rather just the client pool

Reply:
Agreed.

Reply:
@rymurr @rdblue You are right. I changed this part.